kiota_serialization_form/form_serialization_writer.py

```diff
@@ -186,24 +186,23 @@ def write_object_value(
             additional_values_to_merge (tuple[Parsable]): The additional values to merge to the
             main value when serializing an intersection wrapper.
         """
-        if key and (value or additional_values_to_merge):
-            temp_writer = self._create_new_writer()
+        temp_writer = self._create_new_writer()
 
-            if value:
-                self._serialize_value(temp_writer, value)
+        if  value is not None:
+            self._serialize_value(temp_writer, value)
 
-            if additional_values_to_merge:
-                for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
-                    self._serialize_value(temp_writer, additional_value)
-                    if on_after := self.on_after_object_serialization:
-                        on_after(additional_value)
+        for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
+            self._serialize_value(temp_writer, additional_value)
+            if on_after := self.on_after_object_serialization:
```

Comment:
we should check for depth like dotnet does to make sure it only serializes first level objects.